Taps

eduard-watchmaker opened the pull request:
Added TAPS to methylseq

This pull request aim to add support for TAPS data in the nf-core/methylseq pipeline.


We added the subworkflow FASTQ_ALIGN_DEDUP_BWAMEM (bwa mem) to align TAPS data (positive readout / conversion of methylated Cs instead of all unmethylated Cs).

We separated MethylDackel from the alignment subworkflow fastq_align_dedup_bwameth. 

We created a new subworkflow called BAM_METHYLDACKEL with the two different methyldackel processes available.

We added a subworkflow to evaluate methylation of bwa mem or bwa-meth aligned reads called 'BAM_TAPS_CONVERSION' which used different rastair tools.

We added the logic to have data flow from either aligner into either methylation caller.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/methylseq/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/methylseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core pipelines lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

sateeshperi committed:
Merge pull request #558 from nf-core/dev

Dev -> Master 4.1.0

eduard-watchmaker committed:
updated version

eduard-watchmaker committed:
updated version

eduard-watchmaker committed:
updated version

eduard-watchmaker committed:
schema

eduard-watchmaker committed:
path

eduard-watchmaker committed:
path

eduard-watchmaker committed:
working version

eduard-watchmaker committed:
changed fq_align_dedup_bwamem for fq_align_bwa

eduard-watchmaker committed:
samtools change

eduard-watchmaker committed:
taps to nfcore

eduard-watchmaker committed:
add bam to workflow name

eduard-watchmaker committed:
renaming

eduard-watchmaker committed:
changes

eduard-watchmaker committed:
more tests

eduard-watchmaker committed:
comment out

eduard-watchmaker committed:
params

eduard-watchmaker committed:
rm file from test

eduard-watchmaker committed:
rm file from test

eduard-watchmaker committed:
rm file from test

eduard-watchmaker committed:
new parabricks fq2bam

eduard-watchmaker committed:
fix

eduard-watchmaker committed:
add picard groups

eduard-watchmaker committed:
why only one sample

eduard-watchmaker committed:
why only one sample

eduard-watchmaker committed:
why only one sample

eduard-watchmaker committed:
why only one sample

eduard-watchmaker committed:
why only one sample

eduard-watchmaker committed:
why only one sample

eduard-watchmaker committed:
boradcast

eduard-watchmaker committed:
boradcast methyldackel

eduard-watchmaker committed:
updated to fastq_align_dedup_bwamem

eduard-watchmaker committed:
Merge branch 'taps' of github.com:nf-core/methylseq into taps

eduard-watchmaker committed:
missing comma

eduard-watchmaker committed:
rm redundant code

eduard-watchmaker committed:
config fix

eduard-watchmaker committed:
forgot damn config

eduard-watchmaker committed:
more channels

Commit:
new snapshots

eduard-watchmaker committed:
less tests

eduard-watchmaker committed:
fixing new version HS metrics

eduard-watchmaker committed:
ignore multiqc parquet

eduard-watchmaker committed:
input was ahead of time lol

Commit:
gpu snapshhot

Commit:
fix lint issues from new template

eduard-watchmaker committed:
Merge branch 'dev' into taps

sateeshperi commented on docs/output.md:
```diff
@@ -174,6 +215,9 @@ _Note that bismark can use either use Bowtie2 (default) or HISAT2 as alignment t
   - Summary file giving lots of metrics about the aligned BAM file.
 
+**bwa-mem output directory: `results/bwamem/alignments/`**
+# TODO
```
TODO

sateeshperi commented on modules/nf-core/parabricks/fq2bammeth/main.nf:
```diff
@@ -1,54 +1,56 @@
     label 'process_high'
     label 'process_gpu'
+    // needed by the module to work properly can be removed when fixed upstream - see: https://github.com/nf-core/modules/issues/7226
+    stageInMode 'copy'
```
not sure we want to turn this on by default in the pipeline....there could be large files that will quickly cause disk space issues. Users can provide this via their custom config as needed

eduard-watchmaker replied:
This was updated on the original parabricks modules in nf-core, I just updated them to the latest version. I suggest we bring this conversation there if we have concerns.

maxulysse commented on docs/images/taps_metromap.png:
where is the svg version of this file?

maxulysse commented on docs/output.md:
```diff
@@ -196,6 +240,9 @@ This step removes alignments with identical mapping position to avoid technical
   - Log file giving summary statistics about deduplication.
 
+**bwa-mem output directory: `results/bwamem/deduplicated/`**
+# TODO
```
TODO

maxulysse commented on docs/output.md:
```diff
@@ -231,6 +278,9 @@ Filename abbreviations stand for the following reference alignment strands:
   - Methylation statuses in [bedGraph](http://genome.ucsc.edu/goldenPath/help/bedgraph.html) format.
 
+**bwa-mem / TAPS workflow output directory: `results/rastair/`**
+# TODO
```
TODO

maxulysse commented on docs/output.md:
```diff
@@ -241,6 +291,7 @@ BedGraph files are filtered using the BED file passed to `--target_regions_file`
 
 **Bismark output directory: `results/bismark/methylation_calls/bedGraph/`**
 **bwa-meth output directory: `results/methyldackel/`**
+# TODO: implement filtering by `--target_regions_file` in bwa-mem
```
Is this feature to do, or is the documentation to do?

eduard-watchmaker replied:
Not implemented yet so will remove.

maxulysse commented on docs/output.md:
```diff
@@ -254,6 +305,10 @@ BedGraph files are filtered using the BED file passed to `--target_regions_file`
 
+
+### Rastair
+# TODO
```
Todo

maxulysse commented on nextflow_schema.json:
```diff
@@ -413,7 +417,38 @@
+                    "format": "file-path",
+                    "description": "Path to the BWA-MEM index filename base"
+                }
+            }
```
shouldn't that go in the references genome options?

eduard-watchmaker replied:
Editing, thanks

Commit:
PR feedback

Commit:
forgot allOf

eduard-watchmaker commented:
No idea why the linting fails, it works perfectly on my computer using nf-core/tools version 3.5.0.dev0

Commit:
paths for output

sateeshperi committed:
update multiqc module and snapshots

sateeshperi committed:
pre-commit fix all files

eduard-watchmaker commented:
Overall changes in TAPS methylseq PR: https://github.com/nf-core/methylseq/pull/574:

Updated bismark modules

Added rastair modules: https://github.com/nf-core/modules/pull/9065

Removed methyldackel modules from fastq_align_dedup_bwameth subworkflow: https://github.com/nf-core/modules/pull/9068

Added bam_methyldackel subworkflow which runs methyldackel modules: https://github.com/nf-core/modules/pull/9066

Added fasta_index_methylseq subworkflow which indexes genomes for bismark, bwa-meth or bwa mem: https://github.com/nf-core/modules/pull/9067

Added bam_taps_conversion to run rastair modules for TAPS data: https://github.com/nf-core/modules/pull/9104

Added fastq_align_dedup_bwamem subworkflow to run bwa mem (+ gpu parabricks), dedup and run stats: https://github.com/nf-core/modules/pull/9197

Refactored the main METHYLSEQ workflow to add the logic for TAPS, also added changes to metromap and docs

Each of these PRs came with auxiliary modules and conf files when needed, as well as all their tests, meta.yml files and snapshots.
